app/components/Upload/FileUpload.tsx

```diff
@@ -1,72 +1,61 @@
-import { Button } from '@webkom/lego-bricks';
-import { Component } from 'react';
-import { connect } from 'react-redux';
-import { uploadFile } from 'app/actions/FileActions';
+import { Flex, Icon } from '@webkom/lego-bricks';
+import { UploadIcon } from 'lucide-react';
+import { useState, useRef } from 'react';
+import { useAppDispatch } from 'app/store/hooks';
 import styles from './FileUpload.css';
 
-type State = {
-  pending: boolean;
-};
-type Props = {
-  uploadFile: (arg0: { file: File }) => Promise<any>;
+type FileUploadProps = {
+  uploadFile: (arg0: { file: File }) => Promise<{
+    meta: { fileKey: string; fileToken: string };
+  }>;
   onChange: (arg0: string, arg1: string) => void;
   className?: string;
 };
 
-class FileUpload extends Component<Props, State> {
-  input: HTMLInputElement | null | undefined;
-  state = {
-    pending: false,
-  };
-  handleClick = () => {
-    this.input && this.input.click();
-  };
-  handleChange = (e) => {
-    const file = e.target.files[0];
-    this.setState({
-      pending: true,
-    });
-    this.props
-      .uploadFile({
-        file,
-      })
-      .then(({ meta }) => {
-        this.setState({
-          pending: false,
-        });
-        this.props.onChange(meta.fileKey, meta.fileToken);
-      })
-      .catch((error) => {
-        this.setState({
-          pending: false,
-        });
-        throw error;
-      });
+const FileUpload = ({ uploadFile, onChange }: FileUploadProps) => {
+  const [pending, setPending] = useState(false);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const handleClick = () => {
+    inputRef.current && inputRef.current.click();
   };
 
-  render() {
-    return (
-      <div>
-        <Button
-          isPending={this.state.pending}
-          onPress={this.handleClick}
-          className={this.props.className}
-        >
-          Last opp fil
-        </Button>
-        <input
-          ref={(ref) => (this.input = ref)}
-          className={styles.input}
-          onChange={this.handleChange}
-          type="file"
-          accept="application/pdf"
-        />
-      </div>
-    );
-  }
-}
+  const dispatch = useAppDispatch();
 
-const mapDispatchToProps = {
-  uploadFile,
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0];
+    if (file) {
+      setPending(true);
+      dispatch(
+        uploadFile({ file })
+          .then(({ meta }) => {
+            setPending(false);
+            onChange(meta.fileKey, meta.fileToken);
+          })
+          .catch((error) => {
+            setPending(false);
+            throw error;
+          }),
+      );
+    }
+  };
+
+  return (
+    <Flex justifyContent="center">
```

Comment:
Does this need an `alignItems="center"`?

Reply:
Yeah, otherwise the icon will not be centered